src/common/snippets/src/lowered/pass/load_movebroadcast_to_broadcastload.cpp

```diff
@@ -50,7 +47,10 @@ bool LoadMoveBroadcastToBroadcastLoad::run(LinearIR& linear_ir, lowered::LinearI
             const auto& load_parent_node = load_expr->get_input_port_connector(0)->get_source().get_expr()->get_node();
             const auto& outshape = move_broadcast->get_output_partial_shape(0);
             const auto broadcastload = std::make_shared<snippets::op::BroadcastLoad>(load_parent_node, *outshape.rbegin(), load->get_offset());
-            expr_it = linear_ir.replace_with_node({ load_expr, expr }, broadcastload);
+            // insert at position of load_expr. As BroadcastMove and Load will be removed, preserve expr_it.
+            expr_it = std::next(expr_it);
+            linear_ir.replace_with_node({ load_expr, expr }, broadcastload, load_expr->get_loop_ids(), linear_ir.find(load_expr));
+            expr_it = std::prev(expr_it);
```

Comment:
@v-Golubev The story is, we need this change to handle dynamic case. As in this case[ `InsertBroadcastMove` and `LoadMoveBroadcastToBroadcastLoad `](https://github.com/openvinotoolkit/openvino/blob/master/src/common/snippets/src/op/subgraph.cpp#L538-L539) are applied again after `extractLoopInvariants`, where load is extracted. The Broadcast and load is not together, and we would like replace` broadcastmove+load` to `loadBroadcast `at place of `load`. Meanwhile, [`LoadMoveBroadcastToBroadcastLoad `find `BroadcastMove`](https://github.com/openvinotoolkit/openvino/blob/master/src/common/snippets/src/lowered/pass/load_movebroadcast_to_broadcastload.cpp#L25)
in its algo. After replace, we need go to expr_it after BroadcastMove, not load.
Below is an example graph before apply `LoadMoveBroadcastToBroadcastLoad `. We need go to `add `instead of `loopBeginDynamic` in next iteration.

Reply:
![54071b0b-cb9c-4b84-aa64-c24d7882f9ec](https://github.com/openvinotoolkit/openvino/assets/48076916/d3c9166b-3fa4-41ba-b0ed-3e47901462cf)